src/core/models/PathwaysMetric.ts

```diff
@@ -0,0 +1,138 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2021 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import { action, makeObservable, observable, runInAction } from "mobx";
+
+import { parseResponseByFileFormat } from "../../api/metrics";
+import { callMetricsApi } from "../../api/metrics/metricsClient";
+import { ERROR_MESSAGES } from "../../constants/errorMessages";
+import RootStore from "../../RootStore";
+import CoreStore from "../CoreStore";
+import { EnabledFilters } from "../types/filters";
+import { Hydratable, MetricRecord, RawMetricData, TenantId } from "./types";
+
+export type BaseMetricConstructorOptions<RecordFormat extends MetricRecord> = {
+  tenantId?: TenantId;
+  sourceFilename: string;
+  dataTransformer: (d: RawMetricData) => RecordFormat[];
+  rootStore?: CoreStore;
+  enabledFilters: EnabledFilters;
+};
+
+/**
+ * Represents a single dataset backed by our metrics API,
+ * plus any applicable metadata.
+ * This is an abstract class that cannot be instantiated directly!
+ * See subclasses that narrow this base down to a specific metric format.
+ * The preferred way to instantiate `Metric` subclasses is in the
+ * MetricsStore.
+ */
+export default abstract class Metric<RecordFormat extends MetricRecord>
```

Comment:
Nit: since this is the default export, can we make the class name match the file name?
```suggestion
export default abstract class PathwaysMetric<RecordFormat extends MetricRecord>
```

Reply:
Good catch.